File: scratch/inspect_data.py
import json
from supabase import create_client

with open(r'c:\Users\Dhanush\Music\aviatorsclaude\.env', 'r') as f:
    env_content = f.read()

supabase_url = ""
supabase_key = ""
for line in env_content.splitlines():
    if line.startswith("VITE_SUPABASE_URL"):
        supabase_url = line.split("=")[1].strip().strip('"').strip("'")
    if line.startswith("VITE_SUPABASE_ANON_KEY"):
        supabase_key = line.split("=")[1].strip().strip('"').strip("'")

supabase = create_client(supabase_url, supabase_key)
res = supabase.table('candidates').select('*').execute()
candidates = res.data

locations = []
rubrics = []
for c in candidates:
    r1 = c.get('round_1_evaluation') or {}
    if isinstance(r1, list):
        r1 = r1[0] if r1 else {}
        
    r2 = c.get('round_2_evaluation') or {}
    if isinstance(r2, list):
        r2 = r2[0] if r2 else {}
        
    # check location
    loc = c.get('location') or c.get('Location') or r1.get('location') or r1.get('Location')
    locations.append(loc)
    
    # check rubrics
    # is there a rubric field in c, r1, or r2?
    keys = list(c.keys()) + list(r1.keys()) + list(r2.keys())
    rubric_keys = [k for k in keys if 'rubric' in k.lower()]
    for rk in rubric_keys:
        if rk in c: rubrics.append(c[rk])
        if rk in r1: rubrics.append(r1[rk])
        if rk in r2: rubrics.append(r2[rk])
        
    # Maybe "feedback" or "reason"?
    reason = r1.get('reason_for_status') or r1.get('reason')
    if reason:
        rubrics.append(reason)

from collections import Counter
print("Locations:")
for k, v in Counter(locations).most_common(20):
    print(f"{k}: {v}")
    
print("\nRubrics/Reasons:")
for k, v in Counter(rubrics).most_common(20):
    print(f"{k}: {v}")

File: master-portal/inspect_data.js
import { createClient } from '@supabase/supabase-js';
import fs from 'fs';

const envContent = fs.readFileSync('../.env', 'utf8');
let supabaseUrl = '';
let supabaseKey = '';

envContent.split('\n').forEach(line => {
    if (line.startsWith('VITE_SUPABASE_URL=')) supabaseUrl = line.substring(line.indexOf('=') + 1).trim().replace(/['"\r]/g, '');
    if (line.startsWith('VITE_SUPABASE_ANON_KEY=')) supabaseKey = line.substring(line.indexOf('=') + 1).trim().replace(/['"\r]/g, '');
});

const supabase = createClient(supabaseUrl, supabaseKey);

async function checkData() {
    const { data: candidates, error } = await supabase.from('candidates').select('*');
    if (error) {
        console.error(error);
        return;
    }

    const locations = {};
    const rubrics = {};

    candidates.forEach(c => {
        let r1 = c.round_1_evaluation || {};
        if (Array.isArray(r1)) r1 = r1[0] || {};
        let r2 = c.round_2_evaluation || {};
        if (Array.isArray(r2)) r2 = r2[0] || {};

        const loc = c.location || c.Location || r1.location || r1.Location || c.current_location || 'Unknown';
        locations[loc] = (locations[loc] || 0) + 1;

        const reason = r1.reason_for_status || r1.reason || 'None';
        if (reason !== 'None') {
            rubrics[reason] = (rubrics[reason] || 0) + 1;
        }
    });

    console.log("Locations:");
    Object.entries(locations).sort((a,b) => b[1]-a[1]).slice(0, 30).forEach(([k,v]) => console.log(`${k}: ${v}`));

    console.log("\nRubrics/Reasons:");
    Object.entries(rubrics).sort((a,b) => b[1]-a[1]).slice(0, 30).forEach(([k,v]) => console.log(`${k}: ${v}`));
}

checkData();
